pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -0,0 +1,279 @@
+		workerPodSpec.NodeSelector = applySelectors(workerPodSpec.NodeSelector, nodeSelectors[index+1].NodeSelector)
+	}
+}
+
+func cloneSelectors(src map[string]string) map[string]string {
+	dst := make(map[string]string, len(src))
+	for k, v := range src {
+		dst[k] = v
+	}
+	return dst
+}
+
+func (j *RayJob) RestoreNodeAffinity(nodeSelectors []jobframework.PodSetNodeSelector) {
+	// head
+	headPodSpec := &j.Spec.RayClusterSpec.HeadGroupSpec.Template.Spec
+	if !equality.Semantic.DeepEqual(headPodSpec.NodeSelector, nodeSelectors[0].NodeSelector) {
+		headPodSpec.NodeSelector = cloneSelectors(nodeSelectors[0].NodeSelector)
+	}
+
+	// workers
+	for index := range j.Spec.RayClusterSpec.WorkerGroupSpecs {
+		workerPodSpec := &j.Spec.RayClusterSpec.WorkerGroupSpecs[index].Template.Spec
+		if !equality.Semantic.DeepEqual(workerPodSpec.NodeSelector, nodeSelectors[index+1].NodeSelector) {
+			workerPodSpec.NodeSelector = cloneSelectors(nodeSelectors[index+1].NodeSelector)
+		}
+	}
+}
+
+func (j *RayJob) Finished() (metav1.Condition, bool) {
+	condition := metav1.Condition{
+		Type:    kueue.WorkloadFinished,
+		Status:  metav1.ConditionTrue,
+		Reason:  string(j.Status.JobStatus),
+		Message: j.Status.Message,
+	}
+	finished := false
+	switch j.Status.JobDeploymentStatus {
+	case rayjobapi.JobDeploymentStatusComplete, rayjobapi.JobDeploymentStatusFailedJobDeploy, rayjobapi.JobDeploymentStatusFailedToGetOrCreateRayCluster:
+		finished = true
+
+	default:
+		finished = false
+	}
+	return condition, finished
+}
+
+func (j *RayJob) EquivalentToWorkload(wl kueue.Workload) bool {
+	if len(wl.Spec.PodSets) != (len(j.Spec.RayClusterSpec.WorkerGroupSpecs) + 1) {
+		return false
+	}
+
+	podSets := wl.Spec.PodSets
+	headPodSpec := &j.Spec.RayClusterSpec.HeadGroupSpec.Template.Spec
+	if podSets[0].Count != 1 {
+		return false
+	}
+	if !equality.Semantic.DeepEqual(headPodSpec.InitContainers, podSets[0].Template.Spec.InitContainers) {
+		return false
+	}
+	if !equality.Semantic.DeepEqual(headPodSpec.Containers, podSets[0].Template.Spec.Containers) {
+		return false
+	}
+
+	// workers
+	for index := range j.Spec.RayClusterSpec.WorkerGroupSpecs {
+		if podSets[index+1].Count != *j.Spec.RayClusterSpec.WorkerGroupSpecs[index].Replicas {
+			return false
+		}
+
+		workerPodSpec := &j.Spec.RayClusterSpec.WorkerGroupSpecs[index].Template.Spec
+		if !equality.Semantic.DeepEqual(workerPodSpec.InitContainers, podSets[index+1].Template.Spec.InitContainers) {
+			return false
+		}
+		if !equality.Semantic.DeepEqual(workerPodSpec.Containers, podSets[index+1].Template.Spec.Containers) {
+			return false
+		}
+	}
+	return true
+}
+
+func (j *RayJob) PriorityClass() string {
+	if j.Spec.RayClusterSpec != nil {
+		rcs := j.Spec.RayClusterSpec
+		if len(rcs.HeadGroupSpec.Template.Spec.PriorityClassName) != 0 {
+			return rcs.HeadGroupSpec.Template.Spec.PriorityClassName
+		}
+		for wi := range rcs.WorkerGroupSpecs {
```

Comment:
See discussion here https://github.com/ray-project/kuberay/issues/1100, maybe we can query the `ray.io/priority-class-name` nowadays.

Reply:
Seconding this?

Reply:
`ray.io/priority-class-name` is very specific to the volcano integration of the RayCluster which will just force it in the pods spec.  

I don't see the benefit of also checking that especially since it's not populated for RayJobs. On the other hand if  ray-project/kuberay#1100 adds a dedicated field in the spec we should definitely use it.